perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -952,7 +952,7 @@ sub process_report : Private {
     }
 
     # set some simple bool values (note they get inverted)
-    if ($c->stash->{contributing_as_body}) {
+    if ($c->stash->{contributing_as_body} && !$anon_button) {
```

Comment:
This works fine here, but note later on, body takes precedence over anonymous_user for setting the extra metadata of contributed_as. I think instead of this change, at line 951 it should make sure body/another_user are unset (given only one of these three should be set at once, perhaps it should be a single flag, not 3?).